### ISSUE TITLE
[DEMIS2SORMAS] Further extend the prefilling of labMessages from DEMIS

### ISSUE THREAD

**Issue by FredrikSchaefer** [SORMAS-Foundation/SORMAS-Project#4738]
### Situation Description

The Gematik uploaded new information about how demis notifications are structured. It is about how data is represented in the Specimen.Note and Observation.Note which so far were just unspecified free text fields. The information can be found in the lowest two grey boxes [here](https://confluence-demis.gematik.de/download/attachments/76579105/image2021-2-17_15-1-19.png?version=1&modificationDate=1613570479862&api=v2).

With this information, it should be possible to prefill more values in SORMAS, because one can probably extract single variables like `8430 - Probematerial-Bezeichnung` from a demis notification.
At the state of this writing, these are the new variable extracteable:

* Specimen.Note
   1. `8428 - Probematerial-Ident`
   2. `8429 - Probematerial-Index`
   3. `8430 - Probematerialbezeichnung`
   4. `8431 - Probematerialspezifikation`
* Observation.Note
   1. `8420 - Ergebnis-Wert`
   2. `8421 - Einheit`
   3. `8480 - Ergenis-Text`
   4. `8470 - Testbezogene Hinweise`
   5. `8460 - Normalwert-Text`
   6. `8461 - Normalwert-Untergrenze`
   7. `8462 - Normalwert-Obergrenze`
   8. `8422 - Grenzwert-Indikator`

### Feature Description

- [ ] Map the relevant variables by writing them to dedicated fields of a labMessage in the adapter
- [ ] Inplement the automatic transfer of new values for the labMessage processing

### Open Questions
- Which of the new variables are relevant and where should they go in SORMAS?

**Comment by FredrikSchaeferVitagroup** [comment]
@JaquM We probably need your input before we can do anything here.

**Comment by JaquM** [comment]
Both areas seem to be free text fields i.e. annotations. 

Information in Specimen.Note is o communicate any details or issues about the specimen or during the specimen collection. (for example: broken vial, sent with patient, frozen). 
Map them to "Test result details".

Information in Observation.Note are annotations as well. They may include general statemets about observation, or statements about significant or unexpected values. 
Map them to "Test result details" aswell.

**Comment by FredrikSchaeferVitagroup** [comment]
> @FredrikSchaeferVitagroup
> Both areas seem to be free text fields i.e. annotations.
> 
> Information in Specimen.Note is o communicate any details or issues about the specimen or during the specimen collection. (for example: broken vial, sent with patient, frozen).
> Map them to "Test result details".
> 
> Information in Observation.Note are annotations as well. They may include general statemets about observation, or statements about significant or unexpected values.
> Map them to "Test result details" aswell.

@JaquM Yes, they are free text fields, but filled out in a specific way that would allow us to extract information.
I remember the need to prefill PathogenTest.cqValue. When extracting information from those text fields, we could probably do that.

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup Thus far, we've collected information for the mapping of the sampleMaterial. Could you please give us feedback on the comprehensibility of the table and say if we should handle addition of values in another issue or keep them here? 

[MappingDEMIS_210313.xlsx](https://github.com/hzi-braunschweig/SORMAS-Project/files/6134805/MappingDEMIS_210313.xlsx)

**Comment by FredrikSchaeferVitagroup** [comment]
@kwa20 Thank you for the detailled mapping table. We should better move the SampleMaterial mapping to another issue. The smaller the work packages, the better.

Before I create a new issue for that, I'm having two little questions about the mapping table, though:

- The table has  the column DEMIS_CODE. Does this refer to `8428 - Probematerial-Ident` or to `8429 - Probematerial-Index`? I furthermore assume that DEMS_DISPLAY refers to `8430 - Probematerialbezeichnung`.
- Regarding s004, s005, s006 and s0011 it is not really clear to me what to do with them. Shall we map them to SampleMaterial.OTHER and write the DEMS-Display into SampleMaterial.sampleMaterialText, or shall we extend the SampleMaterial enum with the according values you specified in the column `Add SORMAS_VALUES_sampleMaterial`. In the second case we would not need to map them to OTHER, right?

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup the sampleMaterial would be the corresponding mapping to the "Probenmaterial". Do you want to use this as a concept issue and subdivide it?

We had a look at the "simplifier"(https://simplifier.net/guide/implementierungsleitfadenfrdiesars-cov-2-erregernachweismeldung/hintergrund) and under ValueSets we found these values for the material. I assumed that this relates to the specimen.note but honestly didn't find a specific reference for this on the page. 

In regards to s004/5/6/11: We don't have these values yet, so we could map them through Other and then fill the subsequent text for now and re-map them when the values are added.

**Comment by FredrikSchaeferVitagroup** [comment]
@kwa20 

> @FredrikSchaeferVitagroup the sampleMaterial would be the corresponding mapping to the "Probenmaterial". Do you want to use this as a concept issue and subdivide it?

Yes.

> In regards to s004/5/6/11: We don't have these values yet, so we could map them through Other and then fill the subsequent text for now and re-map them when the values are added.

I would prefer to do it thoroughly right away. I've created an issue for that ([4773](https://github.com/hzi-braunschweig/SORMAS-Project/issues/4773)). It does not really go faster to first map it to OTHER.

**Comment by FredrikSchaeferVitagroup** [comment]
The RKI provided more information about the mapping of DEMIS-Values regarding the pathogen test type.
[MappingDEMIS_pathogenTestType.pdf](https://github.com/hzi-braunschweig/SORMAS-Project/files/6241826/MappingDEMIS_pathogenTestType.pdf)
DEMIS seems to be using LOINC-Codes here. 
@kwa20 @JaquM : Maybe that enables you to create some mapping to the PathogenTestResult.PathogenTestType

**Comment by FredrikSchaeferVitagroup** [comment]
Also, they provided information about planned changes in DEMIS [here](https://simplifier.net/demis).
